test/mocks/router/mocks.h

```diff
@@ -39,15 +40,35 @@ class MockRetryState : public RetryState {
   DoRetryCallback callback_;
 };
 
+class TestRateLimitPolicyEntry : public RateLimitPolicyEntry {
```

Comment:
If you mock any functions, just mock them all for consistency. Switch this to MockRateLimitPolicyEntry. Any class that has mocks should have constructor/destructor defined in header and implemented in cc file. Same for TestRateLimitPolicy